contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -2493,6 +2513,157 @@ is_char_identpart(char c)
 			(c >= '0' && c <= '9'));
 }
 
+/*
+ * Put delimiters around a T-SQL variable/parameter that is
+ * named '@@var' or contains a hash, e.g. '@var#'.
+ * Without delimiters, the backend will raise an error.
+ * This is used for the parameter argument of sp_executesql, so the input
+ * string may contain multiple names, e.g.: @par1 int, @par2 varchar(20), ...
+ */
+static char *
+delimit_tsql_atatuservar(const char *src)
+{
+	char *s = (char *) src;
+	char *varname_start = NULL;
+	bool add_delimiter = false;
+	
+	/* 
+	 * Reserving twice the amount of space of the input string: since the shortest possible
+	 * parameter definition is 5 characters ('@@p x' , where x would be the type), this will 
+	 * always be enough for adding delimiters.
+	 * Note that there can be multiple parameter names in the input string.
+	 */
+	char *result = (char *) palloc(sizeof(char)*strlen(src)*2);	
```

Comment:
One option rather than pre-allocating the max possible string is to just use a StringInfo struct, which automatically handles string resizing. See https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_4_X_DEV/contrib/babelfishpg_tsql/src/pl_exec-2.c#L281 for examples on how to use such a struct (in this case you don't need to free the data afterwards, and can just return it)

Reply:
We're talking about pretty small strings, a long string would be 100 bytes but often we see something like 20-30 chars. For such small sizes, it does not seem to be worth it.

Reply:
If you are confident that the string size overhead is small, then I am okay with going with this current implementation.

Reply:
I am, thanks